homeassistant/components/syncthing/sensor.py

```diff
@@ -0,0 +1,226 @@
+"""Support for monitoring the Syncthing instance."""
+
+import logging
+
+import aiosyncthing
+
+from homeassistant.const import CONF_NAME
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import (
+    DOMAIN,
+    FOLDER_PAUSED_RECEIVED,
+    FOLDER_SENSOR_ALERT_ICON,
+    FOLDER_SENSOR_DEFAULT_ICON,
+    FOLDER_SENSOR_ICONS,
+    FOLDER_SUMMARY_RECEIVED,
+    SCAN_INTERVAL,
+    SERVER_AVAILABLE,
+    SERVER_UNAVAILABLE,
+    STATE_CHANGED_RECEIVED,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Syncthing sensors."""
+
+    name = config_entry.data[CONF_NAME]
+    synchting = hass.data[DOMAIN][name]
+
+    try:
+        config = await synchting.system.config()
+        version = await synchting.system.version()
+        dev = []
+
+        for folder in config["folders"]:
+            dev.append(
+                FolderSensor(
+                    hass,
+                    synchting,
+                    name,
+                    folder["id"],
+                    folder["label"],
+                    version["version"],
+                )
+            )
+
+        async_add_entities(dev)
+    except aiosyncthing.exceptions.SyncthingError as exception:
+        raise PlatformNotReady from exception
```

Comment:
```suggestion
async def async_setup_entry(hass, config_entry, async_add_entities):
    """Set up the Syncthing sensors."""

    name = config_entry.data[CONF_NAME]
    syncthing = hass.data[DOMAIN][name]

    try:
        config = await syncthing.system.config()
        version = await syncthing.system.version()
        dev = [
            FolderSensor(
                hass,
                syncthing,
                name,
                folder["id"],
                folder["label"],
                version["version"],
            )
            for folder in config["folders"]
        ]

        async_add_entities(dev)
    except aiosyncthing.exceptions.SyncthingError as exception:
        raise PlatformNotReady from exception
```
fix typo and use list-comp

Reply:
I'll update the PR after some testing